### ISSUE TITLE
No way to edit data property list items

### ISSUE THREAD

**Issue by LeoNatan** [LeoNatan/LNPropertyListEditor#1]
Currently, there is no way to edit `NSData` entries

**Comment by JUSTINMKAUFMAN** [comment]
I was thinking about this one, and a feature that'd really set this library apart from what one normally gets when opening a plist file in Xcode would be support for custom `Codable` types. I imagine this would also be a tricky feature to get right.

Two use cases:

1. Right now, pretty much everything that's not a simple String/Int/Bool value is a Dictionary. But if those dictionaries are actually `Codable` types, it'd be awesome to get the actual type names in the `Type` column and be constrained to those types' properties.

2. W/r/t NSData, if the data represents an image (for example), support for displaying that image in the value column would be pretty neat.

**Comment by LeoNatan** [comment]
`Codable` is a Swift concept and cannot be directly supported, but you should already be able to use the existing facilities to implement exactly what you want. The `dataTransformer` property allows you to set an object responsible for transforming objects before display and before update.

```
@protocol LNPropertyListEditorDataTransformer <NSObject>

@optional

- (NSString*)propertyListEditor:(LNPropertyListEditor *)editor displayNameForNode:(LNPropertyListNode*)node;
- (id)propertyListEditor:(LNPropertyListEditor *)editor transformValueForDisplay:(LNPropertyListNode*)node;
- (id)propertyListEditor:(LNPropertyListEditor *)editor transformValueForStorage:(LNPropertyListNode*)node displayValue:(id)displayValue;

@end
```

So, if you have an object which you can represent using key/value pairs, you can use the above API.

Regarding images (and `NSData` editing in general), I was thinking of expanding the delegate methods to notify the user on editing request, and put a [...] button for such complex objects. Perhaps the system can implement a default hex editor for direct data objects, while users can implement their own, for example, for images.

**Comment by LeoNatan** [comment]
You can seen an example of `dataTransformer` here:

https://github.com/wix/DetoxInstruments/blob/23f073b89a60e1229ba186303f62baee05269db7/DetoxInstruments/DetoxInstruments/Controllers/DTXCookiesEditorViewController.m#L102

**Comment by LeoNatan** [comment]
As mentioned in #6:

> This is exactly the main known limitation right now with the framework. I just haven't had time to do something better. My idea was to perhaps find some open source hex editor that has a simple programming interface and incorporate it. Perhaps something like https://github.com/ridiculousfish/HexFiend
> 
> If you are interested in submitting a PR with this functionality, that would be greatly appreciated.

**Comment by LeoNatan** [comment]
This has been implemented in v2.1.0. Now, it is possible to edit data entries.
